Server/UserManagement/UserManagement.Infrastructure/DataSources/UserDataSource.cs

```diff
@@ -15,13 +18,96 @@ public UserDataSource(IDriver driver)
         _driver = driver;
     }
 
-    public async Task<IReadOnlyList<User>> GetAllAsync()
+    public async Task<IReadOnlyList<User>> GetAllAsync(CancellationToken cancellationToken = default)
     {
-        throw new NotImplementedException();
+        await using var session = _driver.AsyncSession();
+
+        var query =
+            $"MATCH (user:{nameof(User)}) " +
+            $"RETURN user.{nameof(User.Id)} AS {nameof(User.Id)}, " +
+            $"    user.{nameof(User.Email)} AS {nameof(User.Email)}, " +
+            $"    user.{nameof(User.PasswordHash)} AS {nameof(User.PasswordHash)} ";
+
+        var users = new List<User>();
+
+        try
+        {
+            var queryResult = await session.RunAsync(query);
+            var queryResultAsList = await queryResult.ToListAsync(cancellationToken);
+            foreach (var queryResultItem in queryResultAsList)
+            {
+                var userAsJson = JsonSerializer.Serialize(queryResultItem.Values);
+                var user = JsonSerializer.Deserialize<User>(userAsJson);
+                users.Add(user);
+            }
+        }
+        finally
+        {
+            await session.CloseAsync();
+        }
```

Comment:
I think it'll be fine to get the session by public method of the IDisposable class, that will incapsulate connection management, such a CloseAsync etc.

Reply:
Great idea! <3